src/stateProvider.ts

```diff
@@ -0,0 +1,42 @@
+import * as core from "@actions/core";
+
+import { State } from "./constants";
+
+export interface IStateProvider {
+    //setOutput(key: string, value: string): void;
+    setState(key: string, value: string): void;
+    getState(key: string): string;
+
+    getCacheState(): string | undefined;
+}
+
+class StateProviderBase implements IStateProvider {
+    getCacheState(): string | undefined {
+        const cacheKey = this.getState(State.CacheMatchedKey);
+        if (cacheKey) {
+            core.debug(`Cache state/key: ${cacheKey}`);
+            return cacheKey;
+        }
+
+        return undefined;
+    }
```

Comment:
Coming from this [comment](https://github.com/actions/cache/pull/1018/files#r1041929081), 

it seems like this is just a `getState()` implementation with a parameter. We can remove this and depend on `getState` method of `StateProvider` and `NullStateProvider`